### ISSUE TITLE
Clippy can't fix warnings because compiler false error

### ISSUE THREAD

**Issue by asensio-project** [rust-lang/rust#110380]
Hello everyone:

I was trying to fix clippy warnings with: ```cargo clippy --fix```. But I got an error of the compiler that of course doesn't exist. My code works fine.

I tried this code:

```rust
pub async fn register_user(
    user_data: web::Json<UserData>,
    pool: web::Data<Pool>,
) -> Result<HttpResponse, actix_web::Error> {
    let user_data = user_data.into_inner();

    if !(user_data.gender == String::from("B") || user_data.gender == String::from("G")) {
        return Err(actix_web::error::ErrorBadRequest("Missing gender"));
    } else if !(user_data.role == String::from("A") || user_data.role == String::from("T")) {
        return Err(actix_web::error::ErrorBadRequest("Missing role"));
    }

    let user = web::block(move || query(user_data, pool)).await??;

    Ok(HttpResponse::Ok().json(&user))
}
```

I expected to see this happen: Successful. All warnings of clippy fixed.

Instead, this happened:
```
warning: failed to automatically apply fixes suggested by rustc to crate `trivial_packager`

after fixes were automatically applied the compiler reported errors within these files:

  * src/register_handler.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see 
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0425]: cannot find value `user` in this scope
  --> src/register_handler.rs:32:32
   |
32 |     Ok(HttpResponse::Ok().json(user))
   |                                ^^^^ not found in this scope

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
Original diagnostics will follow.
```
My full code can be found at [github](https://github.com/asensio-project/trivial_packager).

### Meta

`rustc --version --verbose`:
```
rustc 1.68.2 (9eb3afe9e 2023-03-27)
binary: rustc
commit-hash: 9eb3afe9ebe9c7d2b84b71002d44f4a0edac95e0
commit-date: 2023-03-27
host: x86_64-unknown-linux-gnu
release: 1.68.2
LLVM version: 15.0.6
```

<details><summary>Backtrace</summary>
<p>

```
The same output as the error.
```

</p>
</details>

**Comment by asensio-project** [comment]
Already reported in [clippy](https://github.com/rust-lang/rust-clippy/issues/10648).

**Comment by Nilstrieb** [comment]
Closing it here then.
